Receiving Unprocessable Entity on Version 1.0.1 when errors are present

# Bug Report 

## Describe the bug

When Rubocop has a violation of a cop I receive Unprocessable Entity error and the lint fails to execute and post back the annotations

# To Reproduce

Steps to reproduce the behavior:

Have a violation in the repo on push.

# Expected behavior

Post annotations for rubocop violations

## Stacktrace

```
/action/lib/github_client.rb:36:in `request_http': Unprocessable Entity: {"message":"Invalid request.\n\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\n\"annotation_level\", \"end_line\", \"message\", \"start_line\" weren't supplied.\nNo more than 50 items are allowed; 54 were supplied.","documentation_url":"https://developer.github.com/v3/checks/runs/#update-a-check-run"} (RuntimeError)
	from /action/lib/github_client.rb:10:in `patch'
	from /action/lib/github_check_run_service.rb:22:in `run'
	from /action/lib/index.rb:29:in `<main>'
##[error]Docker run failed with exit code 1
```

## Action Version & Workflow File
 
 - Version 1.0.1

### Workflow File

```yaml
ame: Linter
on: [push]

jobs:
  ruby:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Rubocop Linter
      uses: andrewmcodes/rubocop-linter-action@v1.0.1
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -53,7 +53,7 @@ class GithubCheckRunService
         title: CHECK_NAME,
         summary: @summary,
         annotations: @annotations
-      }.compact
+      }
     )
   end
 end

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -19,8 +19,6 @@ class GithubCheckRunService
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conclusion(@report)
 
-    puts update_check_payload
-
     @client.patch(
       "#{endpoint_url}/#{id}",
       update_check_payload

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -11,15 +11,15 @@ class GithubCheckRunService
   end
 
   def run
-    id = @client.post(
+    x = @client.post(
       endpoint_url,
       create_check_payload
-    )['id']
+    )
+    puts x
+    id = x['id']
     @summary = @report_adapter.summary(@report)
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conclusion(@report)
-
-    puts update_check_payload
 
     @client.patch(
       "#{endpoint_url}/#{id}",

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -52,7 +52,7 @@ class GithubCheckRunService
       output: {
         title: CHECK_NAME,
         summary: @summary,
-        annotations: (@annotations if @conclusion == 'failure')
+        annotations: @annotations
       }.compact
     )
   end

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -19,6 +19,8 @@ class GithubCheckRunService
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conclusion(@report)
 
+    puts update_check_payload
+
     @client.patch(
       "#{endpoint_url}/#{id}",
       update_check_payload

--- a/lib/github_check_run_service.rb
+++ b/lib/github_check_run_service.rb
@@ -11,15 +11,15 @@ class GithubCheckRunService
   end
 
   def run
-    x = @client.post(
+    id = @client.post(
       endpoint_url,
       create_check_payload
-    )
-    puts x
-    id = x['id']
+    )['id']
     @summary = @report_adapter.summary(@report)
     @annotations = @report_adapter.annotations(@report)
     @conclusion = @report_adapter.conclusion(@report)
+
+    puts update_check_payload
 
     @client.patch(
       "#{endpoint_url}/#{id}",

--- a/lib/report_adapter.rb
+++ b/lib/report_adapter.rb
@@ -44,6 +44,7 @@ class ReportAdapter
           )
         end
       end
+      return annotation_list
     end
 
     private

--- a/lib/report_adapter.rb
+++ b/lib/report_adapter.rb
@@ -44,7 +44,7 @@ class ReportAdapter
           )
         end
       end
-      return annotation_list
+      annotation_list
     end
 
     private

